src/restLevelClient/generateClientDefinition.ts

```diff
@@ -268,6 +272,9 @@ function getOperationReturnType(
     }
   }
 
+  if (returnType === "HttpResponse") {
```

Comment:
Maybe check for HttpResponse to be included in returnType string to cover cases like: `FooResponse | HttpResponse`

Reply:
This one should also be resolved.